addons/point_of_sale/models/pos_config.py

```diff
@@ -613,51 +613,67 @@ def _open_session(self, session_id):
     # is installed, or if POS is installed on database having companies that already have
     # a localisation installed
     @api.model
-    def post_install_pos_localisation(self):
-        self.assign_payment_journals()
-        self.generate_pos_journal()
-
-    @api.model
-    def assign_payment_journals(self, companies=False):
+    def post_install_pos_localisation(self, companies=False):
         self = self.sudo()
         if not companies:
             companies = self.env['res.company'].search([])
-        for company in companies:
-            if company.chart_template_id:
-                cash_journal = self.env['account.journal'].search([('company_id', '=', company.id), ('type', '=', 'cash')], limit=1)
-                pos_receivable_account = company.account_default_pos_receivable_account_id
-                payment_methods = self.env['pos.payment.method']
-                if cash_journal:
-                    payment_methods |= payment_methods.create({
-                        'name': _('Cash'),
-                        'receivable_account_id': pos_receivable_account.id,
-                        'is_cash_count': True,
-                        'cash_journal_id': cash_journal.id,
-                        'company_id': company.id,
-                    })
+        for company in companies.filtered('chart_template_id'):
+            pos_configs = self.search([('company_id', '=', company.id)])
+            pos_configs.setup_defaults(company)
+
+    def setup_defaults(self, company):
```

Comment:
Do you think that all changes made here still needed? I also think that "setup_invoice_journal" is not really needed anymore.

Reply:
Because of this refactoring, we are able to 1) remove the extensions we made in `_create_bank_journals` and `generate_journals` and move them in more centralized location (`setup_defaults`). (Perhaps I should name it `setup_accounting_defaults`.) Another benefit is that, 2) we can now easily define a method that loops thru all the pos.config of a given company and setup anything by default when a localisation is installed by hooking it in the `setup_defaults` method. This includes the `setup_invoice_journal`.

`setup_invoice_journal` function is created because the specs require the invoicing capability to be active by default. Without it, even if the `module_account` is set to default to `True`, there will be no assigned Invoice Journal in the existing pos.config after installing a fiscal localisation.

Reply:
Note that the behavior of the old code didn't change after the refactoring. Except the addition of `setup_invoice_journal` which you can see in this commit, was very easy to integrate: https://github.com/odoo/odoo/pull/51624/commits/3dad52abb1d9c13c60e35388d9123e9d6a2081ca